File: 3066-minimum-operations-to-exceed-threshold-value-ii.cpp
class Solution { 
public: 
    int minOperations(vector<int>& o, int w) { 
        priority_queue<long, vector<long>, greater<long>> j(o.begin(), o.end()); 
        int x = 0;
        while (j.top() < w) { 
            j.push(j.top() * 2 + (j.pop(), j.top())); 
            j.pop(); 
            x++; 
        } 
        return x; 
    }
};

File: 3066-minimum-operations-to-exceed-threshold-value-ii.java
class Solution { 
    public int minOperations(int[] j, int q) { 
        PriorityQueue<Long> u = new PriorityQueue<>(); 
        for (int x : j) u.add((long) x);
        int c = 0; 
        while (u.peek() < q) { 
            u.add(u.poll() * 2 + u.poll()); 
            c++; 
        } 
        return c; 
    }
}
